libs/flexIPC/fleXdIPCConnector/src/FleXdIPCConnector.cpp

```diff
@@ -0,0 +1,260 @@
+#include "BitStream.h"
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <vector>
+#include <cstdint>
+
+namespace {
+    bool checkIfFileExist(const std::string& filePath) {
+        struct stat info;
+        return (stat(filePath.c_str(), &info) == 0);
+    }
+    
+    bool makeDir(const std::string& dirPath) {
+        if (dirPath.empty()) return false;
+        return (mkdir(dirPath.c_str(), S_IRWXU | S_IRWXG | S_IRWXO | S_IXOTH) == 0);
+    }
+    
+    std::string cutLastPathElement(const std::string& dirPath) {
+        return dirPath.substr(0, dirPath.find_last_of("\\/"));
+    }
+    
+    bool makeParentDir(const std::string& filePath) {
+        std::string dir = cutLastPathElement(filePath);
+        if(checkIfFileExist(dir)) return true;
+        if(!checkIfFileExist(cutLastPathElement(dir))) makeParentDir(dir);
+        return makeDir(dir);
+    }
+}
+
+
+namespace flexd {
+    namespace icl {
+        namespace ipc {          
+            IPCConnector::IPCConnector(uint32_t myID, FleXdEpoll& poller, bool garantedDelivery /*false*/)
+            : m_myID(myID),
+              m_poller(poller),
+              m_garantedDelivery(garantedDelivery),
+              m_server(nullptr),
+              m_clients()
+            {
+                makeParentDir(FLEXDIPCUDSPATH + std::to_string(m_myID));
+            }
+            
+            IPCConnector::~IPCConnector()
+            {
+                
+            }
+            
+            uint32_t IPCConnector::getMyID() const {
+                  return m_myID;
+            }
+              
+            bool IPCConnector::addPeer(uint32_t peerID)
+            {
+                const std::string socPath = FLEXDIPCUDSPATH + std::to_string(peerID);
+                if(checkIfFileExist(socPath)) {
+                    return addClient(peerID, socPath);
+                } else {
+                    if(!m_server) {
+                        const std::string myPath = FLEXDIPCUDSPATH + std::to_string(m_myID);
+                        m_server = std::make_shared<FleXdIPCProxyBuilder<FleXdUDSServer> >(myPath, m_poller);
+                        m_server->setOnRcvMsg([this](pSharedFleXdIPCMsg msg, int fd){ this->onRcvMsg(msg, fd); });
+                        m_server->setOnConnectClient([this](int fd){ this->onConnectClient(fd); });
+                        m_server->setOnDisconnectClient([this](int fd){ this->onDisconnectClient(fd); });
+                        if(!m_server->init()) {
+                            return false;
+                        }
+                    }
+                    auto ret = m_clients.insert(std::make_pair(peerID, Client(false, -1, m_server)));
+                    return ret.second;
+                }
+                return false;
+            }
+            
+            bool IPCConnector::removePeer(uint32_t peerID)
+            {
+                if(m_myID != peerID) {
+                    auto it = m_clients.find(peerID);
```

Comment:
Use: 'return m_clients.erase(peerID) != 0;', since '::erase' returns the amount of elements removed (int)